design-proposals/app-orch-deploy-applications.md

```diff
@@ -0,0 +1,209 @@
+# Design Proposal: Deployment of Applications without using a Deployment Package
+
+Author(s): Scott Baker
+
+Last updated: 2025-05-08
+
+## Abstract
+
+Application Orchestration deploys Helm Charts.
+
+A Helm Chart is represented in the App Orch Catalog as an `Application`, with each `Application`
+object referring to exactly one Helm Chart. Applications are further bundled into
+`Deployment Packages` with each `Deployment Package` containing one or more `Applications`. This
+allows a collection of Applications (each of which is a Helm Chart) to be deployed as a unit to
+one or more edge nodes. This feature has been leveraged for complex use cases that are built from
+many helm charts.
+
+![Multi-Chart Application](images/app-orch-deployment-package.png)
+
+While the `Deployment Package` offers flexibility in supporting complex use cases, it also
+leads to unnecessary complexity for most common case, which only involve single a Helm Chart.
+To deploy a single chart, the user has to create both an `Application` and `Deployment Package`.
+The `Deployment Package` provides no added value if it points to exactly one `Application`,
+but it creates a significant amount of additional work. Furthermore, a `Registry` object must
+also exist to abstract the OCI registry URL of the chart.
+
+![Single-Chart Application](images/app-orch-dp-single-chart.png)
+
+This proposal is to allow deployment of `Application` without requiring creation of a
+`Deployment Package`.
+
+## Proposal
+
+### Background
+
+Below is a more detailed breakdown of the catalog data model. Some fields of these objects are
+omitted for brevity, and this proposal will focus only on the fields that are relevant to
+this proposal. Fields like display-names, descriptions, etc., have been omitted.
+
+- `Registry`. A registry represents a Helm Registry. The `Registry` object contains the following:
+
+  - `name`. Name/identifier for the `Registry`.
+  
+  - `kind`. Whether the registry is used for Helm Charts or Docker Images.
```

Comment:
Line 40 says its Helm Registry and here we are saying its Docker and Helm Chart registry. May be line 40 needs update? Or may check other fields of this section. Are the fields generic or specific to helm chart registry?

Reply:
I will update.

Reply:
fixed.